service/controller/machine_deployment.go

```diff
@@ -741,14 +740,15 @@ func newMachineDeploymentResources(config MachineDeploymentConfig) ([]resource.I
 	return resources, nil
 }
 
-func newMachineDeploymentToClusterFunc(g8sClient versioned.Interface) func(ctx context.Context, obj interface{}) (infrastructurev1alpha3.AWSCluster, error) {
+func newMachineDeploymentToClusterFunc(ctrlClient ctrlClient.Client) func(ctx context.Context, obj interface{}) (infrastructurev1alpha3.AWSCluster, error) {
 	return func(ctx context.Context, obj interface{}) (infrastructurev1alpha3.AWSCluster, error) {
 		cr, err := key.ToMachineDeployment(obj)
 		if err != nil {
 			return infrastructurev1alpha3.AWSCluster{}, microerror.Mask(err)
 		}
 
-		m, err := g8sClient.InfrastructureV1alpha3().AWSClusters(cr.Namespace).Get(ctx, key.ClusterID(&cr), metav1.GetOptions{})
+		var m *infrastructurev1alpha3.AWSCluster
+		err = ctrlClient.Get(ctx, client.ObjectKey{Name: cr.Name, Namespace: cr.Namespace}, m)
```

Comment:
With old client the CR name was `key.ClusterID(&cr)` and with the new one it is `cr.Name`. Is that intended?

Reply:
👍🏻 Good catch, I saw that too and currently fixing it.